node/primitives/src/approval.rs

```diff
@@ -77,6 +77,38 @@ pub enum AssignmentCertKind {
 	},
 }
 
+/// Certificate is changed compared to `AssignmentCertKind`:
+/// - introduced RelayVRFModuloCompact
+#[derive(Debug, Clone, Encode, Decode, PartialEq, Eq)]
+pub enum AssignmentCertKindV2 {
+	/// An assignment story based on the VRF that authorized the relay-chain block where the
+	/// candidate was included combined with a sample number.
+	///
+	/// The context used to produce bytes is [`RELAY_VRF_MODULO_CONTEXT`]
+	RelayVRFModulo {
+		/// The sample number used in this cert.
+		sample: u32,
+	},
+	/// Multiple assignment stories based on the VRF that authorized the relay-chain block where the
+	/// candidate was included combined with a sample number.
+	///
+	/// The context used to produce bytes is [`RELAY_VRF_MODULO_CONTEXT`]
+	RelayVRFModuloCompact {
+		/// The number of samples.
+		sample: u32,
+		/// The assigned cores.
+		core_indices: Vec<CoreIndex>,
```

Comment:
As I noted elsewhere, this could disappear too since it's uniquely determined, except you could keep this but permit `core_indices` to be a subset of the cores output by the vrf.

Reply:
This is exactly what happens now, this needs to be a subset and there is a check for that.

Reply:
Oh I messed this check then.